static/js/user_profile.js

```diff
@@ -192,6 +214,62 @@ export function register_click_handlers() {
         e.preventDefault();
     });
 
+    $("body").on("submit", "#user-profile-modal .subscription_list_remove form", (e) => {
+        e.preventDefault();
+        const stream_row = $(e.currentTarget).closest("tr");
+        const stream_id = Number.parseInt(stream_row.attr("data-stream-id"), 10);
+        const sub = sub_store.get(stream_id);
+        const target_user_id = Number.parseInt(
+            stream_row.closest("#user-profile-modal").attr("data-user-id"),
+            10,
+        );
+        const alert_box = $("#streams-tab .stream_list_info");
+
+        function removal_success(data) {
+            if (data.removed.length > 0) {
+                // Remove stream from the list.
+                stream_row.remove();
```

Comment:
This should ideally be replaced by logic in `stream_events.js` to rerender this widget when a user's stream subscriptions change.  We should add a comment here to that effect, and probably open a follow-up issue for implementing that.

(It's not a super high priority, since this isn't a high visibility widget, but we do consider any failure of real-time sync to be a bug).

Reply:
I'd do this by moving "// The rest of the work is done via the subscription -> remove event we will get" to

```
// Most of the work for handling the unsubscribe is done by the subscription -> remove event we will get.
// However, the user profile component has not yet implemented live update, so we do update its 
//  UI manually here.
```

Reply:
This is mostly great, but needs a small tweak with the line spacing -- we want this comment to be just before `stream_row.remove()`, and then a blank line after that, so that it's clear this comment applies to that.  

(And maybe we merge things to say "update its UI manually here by removing the stream from this list" or something).

Reply:
Fixed this comment position and wordings at the end.

Reply:
I think we want the blank line after `stream_row.remove`, not before, since that makes it clear that the comment refers to just that line, not the success display afterwards.  I took care of changing this.